TestFollowerDirect is flaky

## Flaky Test

### Which jobs are failing

`chunks (6, Tools Test)` — `TestRegionTestSuite/TestFollowerDirect` in `tools/pd-ctl/tests/region`.

### CI link

https://github.com/tikv/pd/actions/runs/27186216173/job/80255623202

### Reason for failure (if possible)

```
--- FAIL: TestRegionTestSuite (53.40s)
    --- FAIL: TestRegionTestSuite/TestFollowerDirect (2.90s)
        Error Trace: tools/pd-ctl/tests/region/region_test.go:563
        Error:       "Failed to get region: [404] \"[PD:region:ErrRegionNotFound]region 100 not found\"\n\n" does not contain "\"id\":100"
        Test:        TestRegionTestSuite/TestFollowerDirect
```

`followerDirect` writes region 100 on the leader via `cluster.HandleRegionHeartbeat`, then immediately queries each follower with `pd-ctl region 100 --no-forward`. A follower serves the request from its locally synced region cache (`DirectlyGetRaftCluster().GetRegion`), which is populated asynchronously by the region syncer. The test never waits for the region to propagate, so under CI load a follower can still return `404 region not found` before the syncer delivers region 100, making the `"id":100` assertion fail.

### Anything else

The follower query should be wrapped in `testutil.Eventually` so it retries until the synced region shows up. The test was introduced in #10682.

--- a/tools/pd-ctl/tests/region/region_test.go
+++ b/tools/pd-ctl/tests/region/region_test.go
@@ -19,6 +19,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"strconv"
+	"strings"
 	"testing"
 	"time"
 
@@ -30,6 +31,7 @@ import (
 
 	"github.com/tikv/pd/pkg/core"
 	"github.com/tikv/pd/pkg/response"
+	"github.com/tikv/pd/pkg/utils/testutil"
 	"github.com/tikv/pd/server/api"
 	pdTests "github.com/tikv/pd/tests"
 	ctl "github.com/tikv/pd/tools/pd-ctl/pdctl"
@@ -557,10 +559,13 @@ func (suite *regionTestSuite) followerDirect(cluster *pdTests.TestCluster) {
 			re.NoError(err)
 			re.Contains(string(output), "Failed to get region")
 		} else {
-			output, err := tests.ExecuteCommand(cmd, "-u", serverAddr, "region", "100", "--no-forward")
-			re.NoError(err)
-			outputStr := string(output)
-			re.Contains(outputStr, "\"id\":100")
+			// The region is created on the leader and propagates to followers
+			// asynchronously through the region syncer, so the follower may not
+			// have it yet. Retry until the synced region shows up.
+			testutil.Eventually(re, func() bool {
+				output, err := tests.ExecuteCommand(cmd, "-u", serverAddr, "region", "100", "--no-forward")
+				return err == nil && strings.Contains(string(output), "\"id\":100")
+			})
 		}
 	}
 }

